python_on_whales/client_config.py

```diff
@@ -65,6 +65,7 @@ class ClientConfig:
     compose_project_directory: Optional[ValidPath] = None
     compose_compatibility: Optional[bool] = None
     client_call: List[str] = field(default_factory=lambda: ["docker"])
+    client_type: Literal["docker", "podman", "nerdctl", "unknown"] = "unknown"
```

Comment:
I know enum is better supported than `Literal` in static type checkers, but it's mostly the fault of type checkers. I expect support will get better in the future.  Using strings is more beginner-friend and works the same with a dynamic type checker like beartype.

Reply:
Using enums is more friendly at least to certain users who prefer to use types as guidance for what values are allowed rather than docs/IDE support, but I'm fine with either way.

Reply:
The codebase uses string everywhere for this use case, there are already litterals here and there. So let's use Literal to be consistent with the rest of the codebase.